Snippets and templates (#50)
nunit/nunit-vs-templates: README.md

### Project Templates

| Template                     | Platform                     | Language     | Project Type         |
|------------------------------|------------------------------|--------------|----------------------|

> | NUnit 3 Unit Test Project    | Desktop                      | C#           | Traditional Style(3) |

Reviewer: ```suggestion
| NUnit 3 Unit Test Project    | Desktop                      | C#           | Traditional Style <sup>3</sup> |
```

| NUnit 3 Unit Test Project    | Desktop                      | C#           | Sdk Style            | 

> | NUnit 3 Unit Test Project    | Desktop .Net Core            | C#           | Sdk Style            | 

Reviewer: ```suggestion
| NUnit 3 Unit Test Project    | Desktop .NET Core            | C#           | SDK Style            | 
```

\+ 1 more `Sdk` -> `SDK`

> | NUnit 3 Unit Test Project    | Desktop                      | Visual Basic | Traditional Style    |

Reviewer: ```suggestion
| NUnit 3 Unit Test Project    | Desktop                      | Visual Basic | Traditional Style <sup>3</sup> |
```

\+ 3 more

| NUnit 3 Unit Test Project    | Xamarin Android <sup>1</sup> | C#           | Traditional Style    |
| NUnit 3 Unit Test Project    | Xamarin iOS <sup>1</sup>     | C#           | Traditional Style    |
| NUnit 3 Unit Test Project    | Xamarin UWP <sup>1,2</sup>   | C#           | Traditional Style    |

1. Requires [Xamarin for Visual Studio](https://xamarin.com/visual-studio) be installed.
2. Requires the [Windows 10 SDK](https://dev.windows.com/en-us/downloads/windows-10-sdk) be installed.

> 3. Traditional Style => Old style legacy csproj format 

Reviewer: ```suggestion
3. Legacy csproj format 
```

### Code Snippets

| Snippet         | Shortcut     | Language      | Comment |
|-----------------|--------------|---------------|---------|
| Test Fixture    | ntestfixture | C#            | |
| Test Method     | ntest        | C#            | |
| Test Case       | ntestcase    | C#            | |

> | Test Case Source | ntestcasesource1    | C#  | Form 1 |

Reviewer: Could the method-based one be `ntestcasesource` without the `1` since that's the most common?

Author: Yes, but should then the next start with 2 or 1 ?

Reviewer: I'd say 2 because the names `ntestcasesource` and `ntestcasesource1` feel like they would be synonyms if they both existed.

| Test Case Source | ntestcasesource2    | C#  | Form 2 |
| Test Case Source | ntestcasesource3    | C#  | Form 3 |
| Test Case Source | ntestcasesource4    | C#  | Form 3 with testdata|
| Assert.Multiple | nmultiple | C# | Surround snippet |

> Note:  "Form": Refer to [documentation for details](https://github.com/nunit/docs/wiki/TestCaseSource-Attribute)

Reviewer: ```suggestion
To see test case source forms, refer to the [`TestCaseSource` documentation](https://github.com/nunit/docs/wiki/TestCaseSource-Attribute).
```

We highly recommend that you use the following extensions for Visual Studio when
working on this project.

> - [SideWaffle](http://sidewaffle.com/) (Not needed, now included as nuget package) - Adds the 

Reviewer: ```suggestion
- [SideWaffle](http://sidewaffle.com/) (Not needed, now included as NuGet package) - Adds the 
```
